File: e2e/java/RoundTrip.java
// E2E live round-trip for the Java SDK. Computes the shape hash per fixed route via the SDK's own
// shapeOf/normalizePath, prints it, then POSTs the batch to the LIVE sketches endpoint.
import io.github.eobi.sentinel.NemesisShield;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class RoundTrip {
    public static void main(String[] args) throws Exception {
        String token = System.getenv("NEMESIS_TOKEN");
        if (token == null) token = "";
        NemesisShield s = new NemesisShield(token);
        String[][] routes = {
            {"GET", "/app/incidents/inc_ip_1_2_3_4_1786400000000"},
            {"GET", "/app/network/autogon.ai"},
            {"GET", "/app/applications/f47ac10b-58cc-4372-a567-0e02b2c3d479"},
        };
        StringBuilder arr = new StringBuilder("[");
        for (int i = 0; i < routes.length; i++) {
            String route = NemesisShield.normalizePath(routes[i][1]);
            String hash = s.shapeOf(routes[i][0], routes[i][1], false);
            System.out.println("SHAPE " + routes[i][1] + " route=" + route + " hash=" + hash);
            if (i > 0) arr.append(",");
            arr.append("{\"route\":\"").append(route).append("\",\"method\":\"").append(routes[i][0])
               .append("\",\"authenticated\":false,\"status\":200,\"params\":[],\"shape\":\"")
               .append(hash).append("\"}");
        }
        arr.append("]");
        String body = "{\"sketches\":" + arr + "}";
        HttpClient h = HttpClient.newHttpClient();
        HttpRequest req = HttpRequest.newBuilder(URI.create("https://shield.nemesislabs.xyz/api/v1/sketches"))
            .header("Authorization", "Bearer " + token)
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(body))
            .build();
        HttpResponse<String> res = h.send(req, HttpResponse.BodyHandlers.ofString());
        System.out.println("POST_STATUS " + res.statusCode());
    }
}

File: e2e/php/roundtrip.php
<?php
// E2E live round-trip for the PHP SDK. Builds a real sketch per fixed route via the SDK's own
// buildSketch, prints the shape hash, then POSTs the batch to the LIVE sketches endpoint.
require __DIR__ . '/NemesisShield.php';

$token = getenv('NEMESIS_TOKEN') ?: '';
$endpoint = 'https://shield.nemesislabs.xyz/api/v1/sketches';
$routes = [
    ['GET', '/app/incidents/inc_ip_1_2_3_4_1786400000000'],
    ['GET', '/app/network/autogon.ai'],
    ['GET', '/app/applications/f47ac10b-58cc-4372-a567-0e02b2c3d479'],
];

$sketches = [];
foreach ($routes as $r) {
    $s = NemesisShield::buildSketch($r[0], $r[1], [], false, 200);
    echo "SHAPE {$r[1]} route={$s['route']} hash={$s['shape']}\n";
    $sketches[] = $s;
}

$body = json_encode(['sketches' => $sketches]);
$ctx = stream_context_create([
    'http' => [
        'method' => 'POST',
        'header' => "Authorization: Bearer {$token}\r\nContent-Type: application/json\r\n",
        'content' => $body,
        'ignore_errors' => true,
        'timeout' => 15,
    ],
]);
$resp = @file_get_contents($endpoint, false, $ctx);
$code = 0;
if (isset($http_response_header) && preg_match('/\s(\d{3})\s/', $http_response_header[0], $m)) {
    $code = (int) $m[1];
}
echo "POST_STATUS {$code}\n";
